test/runtests.jl

```diff
@@ -58,8 +75,8 @@ end
     bounding(::Type{>:UnitRange}, v::Vector{<:Integer}) = UnitRange(extrema(v)...)
     # bounding(URT::Type{UnitRange{T}}, v::Vector{<:Integer}) where T<:Integer = URT(extrema(v)...)
 
-    function bounding(::Type{>:UnitRange}, _union::Fix{typeof(union), Tuple{UnitRange{T}, UnitRange{T}}}) where T <: Integer
-        (a, b) = _union.a
+    function bounding(::Type{>:UnitRange}, _union::Fix) where T <: Integer
```

Comment:
Why this change? Seems like it belongs in a separate commit conceptually.

Reply:
No, `@fix` now escapes everything with `Some` so the `Fix` type params needed to be changed. I was too lazy to figure out the most narrow params and just dropped them all.

Reply:
Ah, right. Thanks for explaining that.

One of the things I want to use `FixArgs.jl` for is to represent a deferred or symbolic computation, and in that case there are no free/unbound/holes. This is what these tests demonstrate.

Prior to this change, choosing the signature in was straightforward, in my opinion. But wrapping everything in `Some` may be laborious, as you mention.

In the case where all arguments are fixed, there is no need to represent holes. So I'm wondering if there's a solution that keeps the convenience in that special case, without relying on e.g. traits.

We can always make another macro to facilitate writing these signatures, too. But I would be interested in seeing in there is a simpler solution.

Reply:
I am not sure exactly, what you want to do here. Would a higher order function `lift(f, args::Fix...)::Fix` that takes lazy computations `args` represented by holes free `Fix`es and produces a lazy computation that applies `f` to them?

Reply:
`*(x::Fix{typeof(/), Tuple{N1, D1}}, y::Fix{typeof(/), Tuple{N2, D2}})` is intended to be exactly the same computation as `*(::Rational{T1} , ::Rational{T2})`.

The idea is that `Rational{T}` could, instead of being named `Rational`, be defined in terms of deferred division: `Fix{typeof(/), Tuple{T, T}}`.

The intent is very much for the arguments of `Fix` to be part of the public interface, the same way that it is for `Base.Fix1` and `Base.Fix2`. And by public, I mean insofar as `Fix1`/`Fix2` are not exported and not really public to begin with. But from the perspective of methods that dispatch on `Fix1`/`Fix2`, the arguments definitely are required to be public.

Reply:
Things like
https://github.com/JuliaLang/julia/blob/a47cf0084575a8beba0d2eec1701cfc427af2056/base/strings/search.jl#L22
https://github.com/JuliaLang/julia/blob/16a200709c0d8cb271db8d045d71f54c43d56129/base/array.jl#L1866

Reply:
Thanks, these are good examples.

Reply:
Back to the PR. Do you want me to insert more parameters in `bounding(::Type{>:UnitRange}, _union::Fix)`?

Reply:
Sorry the delay. I thought some more time would help clarify my thinking... I'm not sure it did!

My concern is that the type will become too cumbersome to specify. 

It would go from

`Fix{typeof(union), Tuple{UnitRange{T}, UnitRange{T}}}) where T <: Integer`

to (I didn't test this)

`Fix{typeof(union), Tuple{Some{UnitRange{T}}, Some{UnitRange{T}}}}) where T <: Integer`

and actually for what I intended, it probably should match only empty keyword arguments. The design allows dispatch based on the ordered named and types of keyword argument, but Julia's own multiple dispatch does not. So that means the specified type should really be (again I didn't test)

`Fix{typeof(union), Tuple{Some{UnitRange{T}}, Some{UnitRange{T}}}, NamedTuple{(),Tuple{}}}) where T <: Integer`

So probably for my vision, a macro needs to be used to help a user write this out as something like

```julia
@expandtype  function bounding(::Type{>:UnitRange}, _union::(@FixT union UnitRange{T} UnitRange{T})) where T <: Integer
[...]
end
```

Reply:
I forgot to conclude (sorry a draft got lost), the answer is "yes, please insert more parameters". I expect it to be roughly, if not exactly, what I wrote above.

Thank you!